src/core/js/subviews.js

```diff
@@ -0,0 +1,329 @@
+                this.model = Adapt.findById(context.data._id);
+            }
+
+            if (this.model) return;
+
+            // If no model was found.
+            Adapt.log.warn('SubView: No model found for name="'+this.name+'" id="'+this.id+'"');
+        },
+
+        /**
+         * Equality test for invocation instances
+         * @param  {Invocation}  invocation Compare with
+         * @return {Boolean}
+         */
+        isEqual: function(invocation) {
+            if (this.name !== invocation.name) return false;
+            if (this.id !== invocation.id) return false;
+            if (this.model !== invocation.model) return false;
+            if (!_.isEqual(this.options, invocation.options)) return false;
+            return true;
+        },
+
+        /**
+         * @return {Boolean}  Whether this invocation has instantiated a subview or not
+         */
+        hasInstance: function() {
+            return Boolean(this.instance);
+        },
+
+        /**
+         * Create an instance of the named subview
+         * @return {Object}   an instance of the named subview
+         */
+        createInstance: function() {
+            if (!this.model) {
+                return this;
+            }
+
+            var SubViewClass = Adapt.subviews.get(this.name);
+            if (!SubViewClass) {
+                this.instance = null;
+                return this;
+            }
+
+            var instance = new SubViewClass({
+                attributes: {
+                    "data-adapt-subview": this.uid
+                },
+                model: this.model,
+                options: this.options
+            });
+
+            this.instance = instance;
+            return this;
+        },
+
+        destroy: function() {
+            if (this.hasInstance()) {
+                // Make sure to unset instance as remove causes circular clearup
+                var instance = this.instance;
+                this.instance = null;
+                instance.remove();
+            }
+            this.name = null;
+            this.id = null;
+            this.model = null;
+            this.options = null;
+            this.uid = null;
+        }
+
+    }, {
+
+        uid: 0,
+
+        /**
+         * Generate a unique id for each invocation. This is used to match
+         * placeholders with subviews.
+         * @return {Number}
+         */
+        getNextId: function() {
+            if (this.uid >= Number.MAX_SAFE_INTEGER) {
+                this.uid = 0;
+            } else {
+                this.uid++
+            }
+            return this.uid;
+        }
+
+    });
+
+    /**
+     * Represents the handlebars subview statements from the current Adapt location.
+     */
+    var Invocations = Backbone.Controller.extend({
+
+        items: null,
+
+        initialize: function() {
+            this.items = [];
+            this.updatePlaceholders = _.debounce(this.updatePlaceholders, 1);
```

Comment:
Isn't the wait of a debounce in milliseconds?

Reply:
Yup. This just creates a function which when called multiple times synchronously will callback once on the event queue almost immediately after.